cap/error.go

```diff
@@ -167,3 +168,49 @@ func (err *SupervisorRestartError) KVs() map[string]interface{} {
 
 	return acc
 }
+
+// ErrorToleranceReached is an error that gets reported when a supervisor has
+// restarted a child so many times over a period of time that it does not make
+// sense to keep restarting.
+type ErrorToleranceReached struct {
```

Comment:
I would really use the terminology `RestartToleranceReached`. What if there is an evolution where errors do not result in a restart?